pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/SegmentFetcherFactory.java

```diff
@@ -66,42 +69,33 @@ private static void instantiateSegmentFetcher(String protocol, String className)
   // A config for a protocol may not specify the class name, but specify other configs params (used in init())
   public static void initSegmentFetcherFactory(Configuration segmentFetcherFactoryConfig) {
 
-    // Iterate through the configs to fill the map with any new configured classes or Overridden classes.
+    // Iterate through the configs to instantiate any configured classes first.
     Iterator segmentFetcherFactoryConfigIterator = segmentFetcherFactoryConfig.getKeys();
     while (segmentFetcherFactoryConfigIterator.hasNext()) {
       Object configKeyObject = segmentFetcherFactoryConfigIterator.next();
+      if (!configKeyObject.toString().endsWith(SEGMENT_FETCHER_CLASS_KEY)) {
```

Comment:
Why do we need this? End with a class key seems weird.

Reply:
This is the class config for the protocol. If we don't have this, we instantiate the class multiple times, and see any instantiation logs multiple times.

The first look instantiates any classes configured

the second one instantiates any remaining ones in the default list

and last loop calls init on all classes instantiated.